pkg/builder/in_memory_build_queue_test.go

```diff
@@ -558,7 +504,7 @@ func TestInMemoryBuildQueuePurgeStaleWorkersAndQueues(t *testing.T) {
 	})
 	require.NoError(t, err)
 	_, err = stream3.Recv()
-	require.Equal(t, err, status.Error(codes.FailedPrecondition, "No workers exist for instance \"main\" platform {\"properties\":[{\"name\":\"cpu\",\"value\":\"armv6\"},{\"name\":\"os\",\"value\":\"linux\"}]}"))
+	require.Equal(t, err, status.Error(codes.FailedPrecondition, "No workers exist for instance \"main\" platform "))
```

Comment:
Ah, interesting. We just learned that a nil message marshals to the empty string; not `{}`. That sounds like something that's going to confuse people.

Maybe it would be better to do what you proposed on Slack? Revert the changes to in_memory_build_queue.go. Then change the error path to `platform = &remoteexecution.Platform{}`. That way the resulting string is guaranteed to be non-empty.

Reply:
Now that I think of it, I guess we must also do this to make the scheduler web UI work properly. It becomes hard to click on the platform queue name if it's an empty string.